test/Oscoin/Test/Crypto/Blockchain/Block/Generators.hs

```diff
@@ -125,6 +144,27 @@ genBlockWith parentBlock txs st = do
                }
     pure $ mkBlock header txs
 
+
+-- | Generate an arbitrary child block with arbitrary transactions and
+-- with a correct state hash.
+genEvaledBlock
+    :: ( IsCrypto c
+       , Crypto.Hashable c state
+       , Serialise s
+       , Serialise tx
+       , Arbitrary s
+       )
+    => Block c tx s
+    -> state
+    -> [tx]
+    -> Evaluator state tx output
+    -> Gen (Block c tx s, state)
+genEvaledBlock parentBlock parentState txs evl = do
+    let (_, newState) = evalTraverse evl txs parentState
```

Comment:
I am skeptical of ignoring evaluation errors when the `Evaluator` can be passed in, as it's kind of easy to glance over an invariant you're actually assuming in the test. Could we make it so the caller has to explicitly decide to ignore `EvalError`s?

Reply:
In our current implementation of evaluation blocks that contain transactions are valid blocks and no distinction is made between blocks that contain failing transactions and those that don’t. It therefore makes sense to also generate the full set of valid blocks and not restrict ourselves to blocks that only contain non-failing transactions.

As mentioned before, the current approach is not set in stone. I created #500 to track the discussion.

Reply:
I get that. We are talking about a re-usable generator for tests here, however. For the sake of stating all invariants explicitly, this either needs to disallow passing in an `Evaluator` which can fail, or the property needs to specify that it is explicity not interested in evaluation errors (because "current implementation").

Reply:
One approach could also be to allow the `Evaluator` to only be chosen from a set of known evaluators, each of which has certain properties.

Reply:
Currently there is no use case to having the generators only include non-failing transactions. Because of YAGNI I will not implement this the more sophisticated generators. If we need them we can implement them. I documented the behavior regarding failed transactions though.

Reply:
“Currently”

Reply:
I share @kim 's concern here - I guess one option could be to return the errors in the tuple?

Reply:
To restate the concern: if we want this generator to be general-purpose (which is suggested by its name and location), we should somehow capture the errors somewhere. You could make the same argument for capturing the outputs too in fact :thinking: